src/community/jdbcconfig/src/main/java/org/geoserver/jdbcloader/JDBCLoaderPropertiesFactoryBean.java

```diff
@@ -0,0 +1,255 @@
+            }
+            catch(MalformedURLException e) {
+                //failed, try as a file path
+                File f = new File(propUrl);
+                if (f.canRead() && f.exists()) {
+                    url = DataUtilities.fileToURL(f);
+                }
+            }
+        }
+        catch(Exception e) {
+            LOGGER.log(Level.WARNING, "Error trying to read " + propUrl, e);
+        }
+
+        if (url != null) {
+            LOGGER.info("Configuring jdbcloader from " + url.toString());
+            InputStream in = url.openStream();
+            try {
+                config.load(in);
+            }
+            finally {
+                in.close();
+            }
+            return true;
+        }
+        
+        LOGGER.severe("System property " + pref(CONFIG_SYSPROP) + " specified " + propUrl + " but could not be read, ignoring.");
+        return false;
+    }
+
+    private boolean loadConfigFromDataDir(JDBCLoaderProperties config) throws IOException {
+        Resource propFile = getBaseDir().get(pref(CONFIG_FILE));
+        if (Resources.exists(propFile)) {
+            LOGGER.info("Loading jdbcloader properties from " + propFile.path());
+            InputStream stream = propFile.in();
+            try {
+                config.load(stream);
+                return true;
+            } finally {
+                stream.close();
+            }
+        }
+        return false;
+    }
+
+    void saveConfig(JDBCLoaderProperties config) throws IOException {
+        saveConfig(config, "");
+    }
+
+    private void saveConfig(JDBCLoaderProperties config, String comment) throws IOException {
+        Resource propFile = getBaseDir().get(pref(CONFIG_FILE));
+        
+        try {
+            OutputStream out = propFile.out();
+            try {
+                config.store(out, comment);
+            } finally {
+                out.close();
+            }
+
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Error saving jdbc loader properties to file "
+                + propFile.path(), e);
+            propFile.delete();
+        }
+    }
+
+    private void copyScriptsToDataDir() throws IOException {
+        final Resource scriptsDir = getScriptDir();
+        Class<?> scope = getClass();
+        for (String scriptName : getScripts()) {
+            Resource target = scriptsDir.get(scriptName);
+            if (!Resources.exists(target)) {
+                IOUtils.copy(scope.getResourceAsStream(scriptName), target.out());
+            }
+        }
+    }
+    
+    private void copySampleConfigsToDataDir() throws IOException {
+        final Resource baseDirectory = getBaseDir();
+        for (String sampleConfig : getSampleConfigurations()) {
+            Resource target = baseDirectory.get(sampleConfig);
+            if (!Resources.exists(target)) {
+                IOUtils.copy(Thread.currentThread().getContextClassLoader().getResourceAsStream(sampleConfig), 
```

Comment:
Why not just ```getClassLoader```? Do we expect configs loaded by a ClassLoader with different root to the webapp classes? I do not understand.